appveyor.yml

```diff
@@ -40,6 +40,18 @@ environment:
       PYTHON: "3.5"
       ARCH: "64"
       CONFIGURATION: "Release"
+    - JOB: "Build_Debug"
+      GENERATOR: Visual Studio 14 2015 Win64
+      PYTHON: "3.5"
+      ARCH: "64"
+      CONFIGURATION: "Debug"
+      USE_STATIC_CRT: "ON"
+    - JOB: "Build"
+      GENERATOR: Visual Studio 14 2015 Win64
+      PYTHON: "3.5"
+      ARCH: "64"
+      CONFIGURATION: "Release"
+      USE_STATIC_CRT: "ON"
```

Comment:
New two Appveyor's entries with USE_STATIC_CRT defined (for Debug and Release) are added because different incompatible runtimes are used in both new Appveyor's entries and all libs should be built with exact version of runtime. ( "All modules passed to a given invocation of the linker must have been compiled with the same run-time library compiler option (/MD, /MT, /LD)." From [here](https://msdn.microsoft.com/en-us/library/2kzt1wy3.aspx). Also debug uses unique /MDd and /MTd runtimes.)
Probably, we can left only Debug build with USE_STATIC_CRT enabled, since it builds only Arrow CPP and takes less time. How about this?
And here it's the list of others available Appveyor's build job entries:
- JOB: "Cmake_Script_Tests" // Verification that 3rd party libs are found from toolchain and we don't use default built by us (passing very fast in comparison with others)
- JOB: "Build" , GENERATOR: NMake Makefiles, CONFIGURATION: "Release" // Full Release build with "NMake Makefiles" generator. Probably, we can reduce build time to build only Arrow CPP ?
- JOB: "Build_Debug" GENERATOR: Visual Studio 14 2015 Win64 CONFIGURATION: "Debug" // Debug build of only Arrow CPP with Visual Studio 14 2015 Win64 generator. Makes sure that we don't have issues with CMAKE multiconfiguration (Both Debug and Release configs are available on build time when  Visual Studio 14 2015 Win64 generator is used)
- JOB: "Build" GENERATOR: Visual Studio 14 2015 Win64 CONFIGURATION: "Release" // Full Release build with  "Visual Studio 14 2015 Win64" generator. That's supposed (at least by me and seems in docs as well) as most widely used build case.
- JOB: "Toolchain" GENERATOR: Visual Studio 14 2015 Win64 CONFIGURATION: "Release" // Release build using conda-forge Toolchain and "Visual Studio 14 2015 Win64" generator. We have this to catch toolchain build issues as soon as possible (if something broken in conda-forge)
- JOB: "Build" GENERATOR: Visual Studio 15 2017 Win64 APPVEYOR_BUILD_WORKER_IMAGE: Visual Studio 2017 // Full release build with Visual Studio 2017 compiler. Not sure how this important since msvc compiler for Visual Studio 2017 is pretty similar to VS2015, but Appveyor infrastructure installs latest service packs for  Visual Studio 2017 time to time and might have sense to be sure that we support latest versions of msvc.

With 2 changes mentioned above, current build time (which is pretty long already actually) might not increase a long, but coverage will be a little bit affected.

Reply:
How about building both the static and dynamic versions in the same build job? It's only necessary to build the C++ library, I think, for this. For the Python tests we can use the dynamic version

Reply:
I guess you meant to build static-debug and static-release runtimes versions just in one Appveyor's build entry without Python related part. Sounds great! Thanks